modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/js/ckeditor5/plugins/IconsModifier.tsx

```diff
@@ -0,0 +1,128 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {AlignmentCommand, Plugin} from 'ckeditor5';
+
+import getIcon from '../utils/getIcon';
+
+/**
+ * Mapping between names of CKEditor components and Clay icon symbols
+ */
+const COMPONENT_SYMBOL: Record<string, string | null> = {
+	accessibilityHelp: 'accessibility',
+	alignment: 'align-left',
+	blockQuote: 'quote-left',
+	bold: 'bold',
+	bulletedList: 'list-ul',
+	fontBackgroundColor: 'textbox',
+	fontColor: 'text-color',
+	fontFamily: 'font-family',
+	fontSize: 'font-size',
+	horizontalLine: 'hr',
+	indent: 'indent-more',
+	insertTable: 'table2',
+	italic: 'italic',
+	link: 'link',
+	numberedList: 'list-ol',
+	outdent: 'indent-less',
+	redo: 'redo',
+	removeFormat: 'remove-style',
+	sourceEditing: 'code',
+	strikethrough: 'strikethrough',
+	undo: 'undo',
+};
+
+class IconsModifier extends Plugin {
+	init() {
+		const editor = this.editor;
+
+		const componentFactory = editor.ui.componentFactory;
+
+		const componentNames = Array.from(componentFactory.names());
+
+		componentNames.forEach((componentName: string) => {
+			const symbol = COMPONENT_SYMBOL[componentName];
+
+			if (!symbol) {
+				return;
+			}
+
+			// @ts-ignore
+
+			const component = componentFactory._components.get(
+				componentName.toLowerCase()
+			);
+
+			if (!component) {
+				return;
+			}
+
+			componentFactory.add(componentName, (locale) => {
+				const componentInstance = component.callback(locale);
+
+				const componentType = componentInstance.constructor.name;
+
+				if (componentName === 'alignment') {
```

Comment:
Almost all plugins are updating nicely with approach in this PR, only `alignment` has complications. This is because it does not have a static icon, it changes when you change an item from dropdown. Consequently, it has unique internal structure.